lib/Backends/OpenCL/kernels.cl

```diff
@@ -844,17 +849,17 @@ __kernel void convolutionK(__global float *dest, __global float *src,
 
 __kernel void convolutionW(__global void *mem, cl_uint32_t dest,
                            cl_uint32_t src, cl_uint32_t filter,
-                           cl_uint32_t bias, cl_uint32_t filterSize,
-                           cl_uint32_t stride, PaddingTLBR pads, ShapeNHWC odim,
+                           cl_uint32_t bias, ShapeHW filterSizes,
+                           ShapeHW strides, PaddingTLBR pads, ShapeNHWC odim,
                            ShapeNHWC idim, ShapeNHWC filterDim) {
-  convolutionK(&mem[dest], &mem[src], &mem[filter], &mem[bias], filterSize,
-               stride, pads, odim, idim, filterDim);
+  convolutionK(&mem[dest], &mem[src], &mem[filter], &mem[bias], filterSizes,
```

Comment:
@opti-mix , @ZchiPitt : what is the status of this code? Does it only run for training? Any plans to enable OCLConv for training and remove this?

Reply:
@artemrakhov Yes, there are such plans. But we are not there yet. Once OCLConv can be used for training as well, this code is not needed anymore.

As for testing, one needs to run our tests like "mnist -opencl" which would use training with OpenCL backend and thus exercise the changes. Of course, mnist does not use non-square strides. To test that, one would need to write a new unit tests that uses training & non-square strides and kernels.